### DISCUSSION TITLE
Allow workspace-wide "preamble" for math mode

### DISCUSSION THREAD

**Discussion by darsnack** [notaapp/talk#209] [Feature Requests]
Many of my notes are markdown source files for a larger workflow (e.g. writing a book with Pandoc). In these cases, I usually have a LaTeX "preamble" that defines (or re-defines) math commands based on how I want them to render in the final document. For example, I might define
```latex
\newcommand{\prob}[1]{\mathrm{Pr}\left(#1\right)}
```
for a document where I refer to probabilities of various things many times. It's convenient to be able to write `\prob{X > 10}` instead of the longer version.

Unfortunately, these commands are custom, so they show up as errors in the Nota previews. KaTeX supports [defining custom macros](https://katex.org/docs/supported.html#macros) which can be used to setup these commands. It would be convenient to be able to setup the macros passed to KaTeX on a per-workspace basis so that my Nota preview and final document look alike.

**Comment by astoilkov** [comment]
Thanks for the idea. We will explore it and see what we can do.
